build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleJavadocsPlugin.java

```diff
@@ -145,4 +149,62 @@ private void generateJavadocs(Project project, ProjectLayout layout, TaskContain
             task.getReports().getXml().getOutputLocation().set(new File(checkstyle.getReportsDir(), "checkstyle-api.xml"));
         });
     }
+
+    /**
+     * Used to bridge Gradle 8 and Gradle 9 APIs for Gradleception.
+     *
+     * TODO: Remove after Gradle 9
+     */
+    private static class JavadocSupport {
```

Comment:
🤔 Wonder if some of the methods here we could extract into something reusable for plugin authors as a separate library they could use. I see a lot of bad stuff that could come out of this, as it would be a powerful tool, but maybe we can keep it from being overused somehow...

Reply:
This is something we can discuss. I guess this gives some backward compatibilty to plugin authors. Technically we could also generate such wrapper for every upgraded type via annotation processor. 

Not sure if we need that now or only when we get some feedback from users that this is needed